src/python/pants/engine/legacy/address_mapper.py

```diff
@@ -75,7 +76,10 @@ def _internal_scan_specs(self, specs, fail_fast=True, missing_is_fatal=True):
 
     addresses = set()
     for (spec, _), state in root_entries.items():
-      if missing_is_fatal and not state.value.dependencies:
+      if isinstance(state, Throw) and missing_is_fatal:
```

Comment:
This isn't a specific enough match... an exception could be thrown for multiple reasons.

Might need to do an exception type match here?